frontend/src/ui/components/HeroPreview/HeroPreview.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import './styles.scss';
+
+interface HeroPreviewProps {
+  homePageHeroTitleColor: undefined | string;
+  homePageHeroSubtitleColor: undefined | string;
+  homepageOverlayHtml: undefined | string;
+  heroCoverImage: undefined | string;
+}
+
+export const HeroPreview: React.FC<HeroPreviewProps> = (
+  props: HeroPreviewProps
+) => {
+  let titleStyles = {};
+  let subtitleStyles = {};
+  let containerStyles = {};
+  let matched;
+  const heroTextRegex = /^<h1>(.*)<\/h1><p>(.*)<\/p>$/;
+
+  if (props.homePageHeroTitleColor) {
+    titleStyles = {
+      color: props.homePageHeroTitleColor
+    };
+  }
+  if (props.homePageHeroSubtitleColor) {
+    subtitleStyles = {
+      color: props.homePageHeroSubtitleColor
+    };
+  }
+  if (props.homepageOverlayHtml) {
+    matched = heroTextRegex.exec(props.homepageOverlayHtml as string);
+  }
+  if (props.heroCoverImage) {
+    containerStyles = {
+      backgroundImage: `url("${props.heroCoverImage}")`,
+      backgroundSize: '731px 258px'
```

Comment:
@Agrendalath, 

> Should we move backgroundSize to the scss file?

The reason I added it here because the `background-size` property is needed only when the `background-image` property is set. But I can add it to a CSS class and then conditionally add it to the element.

>Also wouldn't it make sense to use something like this instead?
> ```
> background-position: center;
> background-size: cover;
>```
>
> The cover would be a bit more accurate to what happens on LMS, no?
> Also currently there is a repeat effect is you apply an image that does not match the >expected aspect ratio. It doesn't look elegant.

The issue here is that the preview component doesn't have the same dimensions as the LMS hero section. So there is no way to accurately represent how the image of a given size will look in the LMS. While centering the background image and making it stretch to fit will work better here, it will likely not match how it will look on the instance itself.

Do you have any suggestions for dealing with this issue?

Reply:
@lgp171188,

> The reason I added it here because the background-size property is needed only when the background-image property is set. But I can add it to a CSS class and then conditionally add it to the element.

As far as I'm aware, moving them to CSS won't have any downsides and we'll have more organized styling, so +1 from me.

> The issue here is that the preview component doesn't have the same dimensions as the LMS hero section. So there is no way to accurately represent how the image of a given size will look in the LMS. While centering the background image and making it stretch to fit will work better here, it will likely not match how it will look on the instance itself.

LMS (and preview component) will also have different dimensions on different screens (e.g. I'm personally using vertical WQHD or horizontal 4k for this), so using fixed background size will not be deterministic, unless we enforce users to upload image with specific proportions or allow them to crop it by themselves. This is how it looks for me - you can see the repeated part on the bottom:
![image](https://user-images.githubusercontent.com/3189670/78539027-326cbf00-77f2-11ea-867c-9be24210f8db.png)

I think that the best thing at the moment would be having it with `center` and `cover`, just like in LMS. We could tweak this later, if needed.

Reply:
@Agrendalath, I have set the background position to `center` and the size to `cover`.